diff --git a/libavcodec/nvenc.c b/libavcodec/nvenc.c
index 5efd21cf..e14b0e1c 100644
--- a/libavcodec/nvenc.c
+++ b/libavcodec/nvenc.c
@@ -327,11 +327,14 @@ static av_cold int nvenc_load_libraries(AVCodecContext *avctx)
                "Required: %d.%d Found: %d.%d\n",
                NVENCAPI_MAJOR_VERSION, NVENCAPI_MINOR_VERSION,
                nvenc_max_ver >> 4, nvenc_max_ver & 0xf);
-        nvenc_print_driver_requirement(avctx, AV_LOG_ERROR);
-        return AVERROR(ENOSYS);
+        nvenc_print_driver_requirement(avctx, AV_LOG_WARNING);
+        av_log(avctx, AV_LOG_WARNING,
+               "Older NVENC driver detected (API %d.%d); compiled for %d.%d. "
+               "Continuing with driver's supported feature set.",
+               nvenc_max_ver >> 4, nvenc_max_ver & 0xf,
+               NVENCAPI_MAJOR_VERSION, NVENCAPI_MINOR_VERSION);
     }
 
     dl_fn->nvenc_funcs.version = NV_ENCODE_API_FUNCTION_LIST_VER;
-
     err = dl_fn->nvenc_dl->NvEncodeAPICreateInstance(&dl_fn->nvenc_funcs);
     if (err != NV_ENC_SUCCESS)
